### ISSUE TITLE
Excluded dependencies may revive when depended from another artifact

### ISSUE THREAD

**Issue by dmikurube** [embulk/gradle-embulk-plugins#86]
Consider a library `lib-z` has a dependency on `lib-y`, and the `lib-y` has a dependency on `lib-x`.

When an Embulk plugin project has a dependency that has a transitive dependency excluded :

```
// Project: embulk-input-a
apply plugin: "org.embulk.embulk-plugins"

dependencies {
    implementation("com.example:lib-z:0.1.2") {
        exclude group: "com.example", module: "lib-x"
    }
}
```

its `pom.xml` would be like below. `lib-y` is added by the Gradle plugin `org.embulk.embulk-plugins`.

```
<?xml version="1.0" encoding="UTF-8"?>
<project ...>
  ...
  <dependencies>
    <dependency>
      <groupId>com.example</groupId>
      <artifactId>lib-z</artifactId>
      <version>0.1.2</version>
      <scope>compile</scope>
      <exclusions>
        <exclusion>
          <artifactId>com.example</artifactId>
          <groupId>lib-x</groupId>
        </exclusion>
      </exclusions>
    </dependency>
    <dependency>
      <groupId>com.example</groupId>
      <artifactId>lib-y</artifactId>
      <version>0.2.4</version>
      <scope>runtime</scope>
    </dependency>
  </dependencies>
</project>
```

This is okay as an Embulk pluin because Embulk loads only the first-level dependencies. But, it has a problem when another Embulk plugin depends on this Embulk plugin.

```
// Project: embulk-input-b
apply plugin: "org.embulk.embulk-plugins"

dependencies {
    implementation("com.example:embulk-input-a:0.3.6")
}
```

It would have a dependency on `lib-x` because `lib-y` depends on `lib-x`, but that is not as intended.

----

See: https://stackoverflow.com/questions/547805/exclude-all-transitive-dependencies-of-a-single-dependency/7556707#7556707

**Comment by dmikurube** [comment]
We may just want to add `exclude *:*` for additional dependencies expanded from transitive dependencies.

**Comment by dmikurube** [comment]
Fixed in v0.5.0 by making the direct pom.xml manipulation default.
